pkg/trace/force_sampler.go

```diff
@@ -57,3 +58,53 @@ func sampleAt(ctx context.Context, rate uint) context.Context {
 	}
 	return ctx
 }
+
+type otelForceSampler struct {
+	sampleRate uint
+}
+
+func (s *otelForceSampler) Description() string {
+	return "Samples at the specified rate or forces sampling based on the `force_trace` attribute."
+}
+
+//nolint:gocritic // Why: Required to pass SamplingParameters as a copy
+func (s *otelForceSampler) ShouldSample(p sdktrace.SamplingParameters) sdktrace.SamplingResult {
+	psc := oteltrace.SpanContextFromContext(p.ParentContext)
+	sampler, err := sample.NewDeterministicSampler(s.sampleRate)
+	if err != nil {
+		panic(err)
+	}
+
+	var forceTrace bool
+	for _, a := range p.Attributes {
```

Comment:
I'll probably regret this question, but out of curiosity:

Why do we need to cast `a.Key` to a `string`?  Would it be possible to do something like `p.Attributes[fieldForceTrace]` and skip iterating through all the keys like this?  Is `p.Attributes` not a simple `map[string]Value`?

Reply:
Unfortunately it's not. And I really don't like this either.

`p.Attributes` is actually a `[]attribute.KeyValue` which has `Key` and `Value` types with their own wrappers for Go primitive types like `String`, `Bool`, etc.

That makes the api pretty ugly to work with when we need to pull off some values.